packages/nx/src/config/workspaces.ts

```diff
@@ -603,24 +603,50 @@ export function getGlobPatternsFromPackageManagerWorkspaces(
   root: string
 ): string[] {
   try {
-    try {
-      const obj = yaml.load(readFileSync(join(root, 'pnpm-workspace.yaml')));
-      return normalizePatterns(obj.packages || []);
-    } catch {
-      const { workspaces } = readJsonFile<PackageJson>(
-        join(root, 'package.json')
-      );
-      const res = normalizePatterns(
-        Array.isArray(workspaces) ? workspaces : workspaces?.packages ?? []
-      );
-      if (res.length > 0) return res;
+    const patterns: string[][] = [];
+    const packageJson = readJsonFile<PackageJson>(join(root, 'package.json'));
+
+    patterns.push(
+      normalizePatterns(
+        Array.isArray(packageJson.workspaces)
+          ? packageJson.workspaces
+          : packageJson.workspaces?.packages ?? []
+      )
+    );
 
-      const { packages } = readJsonFile<any>(join(root, 'lerna.json'));
-      return packages.length > 0 ? normalizePatterns(packages) : ['packages/*'];
+    if (existsSync(join(root, 'pnpm-workspace.yaml'))) {
+      try {
+        const obj = yaml.load(readFileSync(join(root, 'pnpm-workspace.yaml')));
+        patterns.push(normalizePatterns(obj.packages || []));
+      } catch (e: unknown) {
+        output.warn({
+          title: `${NX_PREFIX} Unable to parse pnpm-workspace.yaml`,
+          bodyLines: [e.toString()],
+        });
+      }
     }
-  } catch {
-    return undefined;
-  }
+
+    if (existsSync(join(root, 'lerna.json'))) {
+      try {
+        const { packages } = readJsonFile<any>(join(root, 'lerna.json'));
+        patterns.push(packages?.length > 0 ? normalizePatterns(packages) : []);
```

Comment:
IT has to be packages/*. I believe hta'ts the default lerna uses

Reply:
Default wasn't mentioned it lerna's docs, I'll add it here, moving it from below.